### PR TITLE
Dynamic Dashboard: Handle hiding cards from the ellipsis menu

### LINKED ISSUE #12507: Handle hide dashboard cards and migrate away from UserDefaults

### PR DESCRIPTION
**Pull request by itsmeichigo**
Closes: #12507 

## Description

This PR handles the loading and hiding of dashboard cards to migrate away from the use of UserDefaults. Changes include:
- Added `canShowOnDashboard` for onboarding and Blaze cards to display them as unavailable if necessary.
- Added callback closures for all dashboard card views to update the cards' visibility when dismissing them from the ellipsis menu.
- Updated the layout of the onboarding card to be consistent with the new design.

## Testing instructions

- Enable the feature flag `dynamicDashboard` and build the app.
- Log in to a store with incomplete onboarding tasks. 
- Enable the onboarding card and confirm that the layout of the card is correct.
- Hide any of the card on the dashboard screen, confirm that they disappear from the screen afterward.
- Tap the Settings icon on the top right of the dashboard screen, confirm that you can enable any card to make them show up again.
- Switch to a store eligible for Blaze or completed all onboarding tasks.
- Tap the Settings icon on the top right of the dashboard screen, confirm that either Blaze or onboarding card is marked as unavailable.

Optional: disable the `dynamicDashboard` feature flag and confirm that hiding the Blaze and onboarding cards still works.

## Screenshots

---
- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

### CODE REVIEW

**Comment by dangermattic** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          1 Warning
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">View files have been modified, but no screenshot or video is included in the pull request. Consider adding some for clarity.</td>
    </tr>
  </tbody>
</table>
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Message">
          1 Message
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:book:</td>
      <td data-sticky="false">This PR is still a Draft: some checks will be skipped.</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_pr-check">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

File: WooCommerce/WooCommerceTests/ViewRelated/Dashboard/Blaze/BlazeCampaignDashboardViewModelTests.swift [hunk 1]
```diff
@@ -8,7 +8,6 @@ import WordPressAuthenticator
 
 /// Test cases for `BlazeCampaignDashboardViewModel`.
 ///
-@MainActor
```
Comment (itsmeichigo) [inline WooCommerce/WooCommerceTests/ViewRelated/Dashboard/Blaze/BlazeCampaignDashboardViewModelTests.swift:11]: Marking the whole test case with `@MainActor` causes a warning about concurrency (which will be an error in Swift 6). So I removed this and added the attribute to all the async tests instead.

File: WooCommerce/Classes/ViewRelated/Dashboard/Onboarding/StoreSetupProgressView.swift [hunk 2]
```diff
@@ -54,13 +54,18 @@ struct StoreSetupProgressView: View {
...
+                    } else {
+                        showingHideStoreSetupListAlert = true
+                    }
                 }
                 .renderedIf(isHideStoreOnboardingTaskListFeatureEnabled)
             } label: {
-                Image(uiImage: .ellipsisImage)
-                    .flipsForRightToLeftLayoutDirection(true)
-                    .foregroundColor(Color(.textTertiary))
+                Image(systemName: "ellipsis")
```
Comment (pmusolino) [inline WooCommerce/Classes/ViewRelated/Dashboard/Onboarding/StoreSetupProgressView.swift:65]: nit: I noticed we already have in the codebase an image from Gridicons called `ellipsis`. Is there any reason why we are using the system one? Is it part of Joe's design?

**Review by pmusolino — APPROVED** [review]
It looks good to me :shipit: 

I just found some minor/unaxpected things (that can be handled also in separate PRs).

#### There is no empty state when there are no cards. Is it also expected to not have at least one card? 



Looking at this state, it seems that we allow to display at least one card, so the above state it's unexpected. You can replicate it, just hiding the last card from the 3 dots.






#### The last card in the list, doesn't have any padding from the bottom side.

**Comment by dangermattic** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          1 Warning
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">This PR is assigned to the milestone <a href="https://github.com/woocommerce/woocommerce-ios/milestone/196">18.3</a>. This milestone is due in less than 2 days.<br />
Please make sure to get it merged by then or assign it to a milestone with a later deadline.</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_pr-check">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

File: WooCommerce/Classes/ViewRelated/Dashboard/Onboarding/StoreSetupProgressView.swift (hunk 2 above)
Comment (itsmeichigo) [reply WooCommerce/Classes/ViewRelated/Dashboard/Onboarding/StoreSetupProgressView.swift:65]: Yes, this was stated explicitly by Joe in the design:

image

Also, system images can be scaled properly so when a device uses a large font, the icons can be scaled accordingly too. This is a recommended use of icons by Apple.

**Comment by itsmeichigo** [comment]
Thanks @pmusolino for the reviews! 

Regarding hiding all the cards from the dashboard - I started a thread [p1713495594539859-slack-C03L1NF1EA3] to confirm the solution but going ahead with hiding the option to dismiss the last card in 67ac31914cef7efa912ce88f4356f3763b4cae13. I'll add any further update if the team comes up with a better solution.

> The last card in the list, doesn't have any padding from the bottom side.

Hafiz took care of this in his PR and I just merged it back to my branch, now every card should have proper padding.

I'm enabling auto-merge for now to unblock development.

**Pull request merged**